File: apps/go/routes/body/json.go
package main

import "github.com/primate-run/go/route"

var _ = route.Post(func(request route.Request) any {
	json, err := request.Body.JSON()
	if err != nil {
		return map[string]any{"error": err.Error()}
	}
	return json
})

File: apps/vue/routes/route-client/json.ts
import response from "primate/response";
import route from "primate/route";

export default route({
  get(request) {
    const toplevel = request.query.has("top-level") ? "TopLevel/" : "";
    return response.view(`RouteClient/${toplevel}JSON.vue`);
  },
  post: route.with({ contentType: "application/json" }, request => {
    return request.body.json() as any;
  }),
});
